pkg/vm/engine/tae/containers/types.go

```diff
@@ -87,10 +87,11 @@ type Vector interface {
 }
 
 type Batch struct {
-	Attrs   []string
-	Vecs    []Vector
-	Deletes *roaring.Bitmap
-	Nameidx map[string]int
+	Attrs    []string
+	Vecs     []Vector
+	Deletes  *roaring.Bitmap
+	Nameidx  map[string]int
+	borrowed bool
```

Comment:
I think Batch doesn't need to know whether the vector is borrowed. For Batch, it just assembles several vectors for use and closes them all at last.